operations/observability/mixins/meta/rules/components/server/alerts.libsonnet

```diff
@@ -49,6 +49,120 @@
+          * {
+          *   alert: 'db-sync not running',
+          *   expr: 'sum (kube_pod_status_phase{pod=~"db-sync.*"}) by (pod) < 1',
+          *   'for': '5m',
+          *   labels: {
+          *     // sent to the team internal channel until we fine tuned it
+          *     severity: 'warning',
+          *     team: 'webapp'
+          *   },
+          *   annotations: {
+          *     runbook_url: 'https://github.com/gitpod-io/runbooks/blob/main/runbooks/DbSyncNotRunning.md',
+          *     summary: 'The db-sync pod is not running. Investigation required.',
+          *     description: 'db-sync pod not running',
+          *   },
+          * },
+          *
+          */
+          {
+            alert: 'MessagebusNotRunning',
+            expr: 'up{job="messagebus"} < 1',
+            'for': '30s',
+            labels: {
+              // sent to the team internal channel until we fine tuned it
+              severity: 'warning',
+              team: 'webapp'
+            },
+            annotations: {
+              runbook_url: 'https://github.com/gitpod-io/runbooks/blob/main/runbooks/MessagebusNotRunning.md',
+              summary: 'The messagebus pod is not running. Investigation required.',
+              description: 'Messagebus pod not running',
+            },
+          },
+          {
+            alert: 'WebAppServicesHighMemoryUsage',
+            // Reasoning: high rates of RAM consumption should only be temporary. Values based on past data (around 5-10 is constant)
+            expr: 'sum(rate(container_memory_working_set_bytes{container!="POD", node=~".*", pod=~"(server|ws-manager-bridge|usage)-.*"}[30m])) by (pod, node) > 10000000',
+            'for': '15m',
+            labels: {
+              // sent to the team internal channel until we fine tuned it
+              severity: 'warning',
+              team: 'webapp'
+            },
+            annotations: {
+              runbook_url: 'https://github.com/gitpod-io/runbooks/blob/main/runbooks/WebAppServicesHighMemoryUsage.md',
+              summary: 'WebApp services consume excessive amounts of memory. Investigation required.',
+              description: 'WebApp Services execcisve memory usage',
+            },
+          },
+          {
+            alert: 'WebAppServicesHighCPUUsage',
+            // Reasoning: high rates of CPU consumption should only be temporary.
+            expr: 'sum(rate(container_cpu_usage_seconds_total{container!="POD", node=~".*", pod=~"(content-service|dashboard|db|db-sync|messagebus|payment-endpoint|proxy|server|ws-manager-bridge|usage)-.*"}[5m])) by (pod, node) > 0.80',
+            'for': '10m',
+            labels: {
+              // sent to the team internal channel until we fine tuned it
+              severity: 'warning',
+              team: 'webapp'
+            },
+            annotations: {
+              runbook_url: 'https://github.com/gitpod-io/runbooks/blob/main/runbooks/WebAppServicesHighCPUUsage.md',
```

Comment:
What would the runbook contain? In a way, services would ideally run at 100% utilization all the time, that way there's nothing wasted.

I'm asking because this tends to be notoriously difficult to make actionable. One could argue that we should instead set CPU/Memory limits on pods to force them to crash and restart (and alert on the restarts) rather than alerting for high utilization.

Reply:
I understand that there are better ways to setup alerts if you go by the book. But looking at past incidents I deem this a pretty good indicator that somthing is fishy. Especially after a recent deployment.

The runbook will never be substantial. This can serve as a first step, and if we find better measures, replace this error with something that make more sense. But for now we have something, at all. :upside_down_face: